core/src/main/java/org/jdbi/v3/core/statement/SqlStatement.java

```diff
@@ -1284,15 +1311,17 @@ public final This bindList(String key, Object... values) {
      *     .list();
      * </pre>
      *
-     * @param key    attribute name
-     * @param values list of values that will be comma-spliced into the defined attribute value.
+     * @param key     attribute name
+     * @param onEmpty handler for null/empty list
+     * @param values  list of values that will be comma-spliced into the defined attribute value.
      * @return this
      * @throws IllegalArgumentException if the list is empty.
+     * @see BindListEmptyHandling
      */
-    public final This bindList(String key, List<?> values) {
-        if (values.isEmpty()) {
-            throw new IllegalArgumentException(
-                    getClass().getSimpleName() + ".bindList was called with an empty list.");
+    public final This bindList(String key, BiConsumer<SqlStatement, String> onEmpty, List<?> values) {
+        if (values == null || values.isEmpty()) {
```

Comment:
Hm, this is a sneaky change.  I guess that's fine to catch `null` here too.

Reply:
Yeah I figured this would be ok to change since nobody is eager for NPEs from libraries